gateleen-queue/src/test/java/org/swisspush/gateleen/queue/queuing/splitter/QueueSplitterImplTest.java

```diff
@@ -0,0 +1,247 @@
+package org.swisspush.gateleen.queue.queuing.splitter;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.impl.headers.HeadersMultiMap;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.swisspush.gateleen.core.configuration.ConfigurationResourceManager;
+import org.swisspush.gateleen.core.configuration.ConfigurationResourceObserver;
+import org.swisspush.gateleen.core.storage.MockResourceStorage;
+import org.swisspush.gateleen.core.util.ResourcesUtils;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.*;
+import static org.swisspush.gateleen.core.configuration.ConfigurationResourceManager.CONFIG_RESOURCE_CHANGED_ADDRESS;
+
+@RunWith(VertxUnitRunner.class)
+public class QueueSplitterImplTest {
+
+    private Vertx vertx;
+    private MockResourceStorage storage;
+    private final String configResourceUri = "/queueSplitters";
+    private ConfigurationResourceManager configurationResourceManager;
+    private QueueSplitterImpl queueSplitter;
+
+    private final String CONFIG_RESOURCE_VALID = ResourcesUtils.loadResource("testresource_queuesplitter_configuration_valid", true);
+    private final String CONFIG_RESOURCE_INVALID = ResourcesUtils.loadResource("testresource_queuesplitter_configuration_invalid", true);
+
+    @Before
+    public void setUp() {
+        vertx = Vertx.vertx();
+        storage = new MockResourceStorage();
+        configurationResourceManager = new ConfigurationResourceManager(vertx, storage);
+        queueSplitter = new QueueSplitterImpl(configurationResourceManager, configResourceUri);
+    }
+
+    @Test
+    public void splitWithMissingConfigResource(TestContext context) {
+
+        // Given
+        Async async = context.async();
+        context.assertFalse(queueSplitter.isInitialized());
+
+        // When
+        queueSplitter.initialize().onComplete(event -> {
+
+            // Then
+            context.assertFalse(queueSplitter.isInitialized());
```

Comment:
Shouldn't the `queueSplitter` be initialized now?

Reply:
If the configuration is missing or invalid the QueueSplitterIml is not initialized